tasks/tenant/update-manager-image-in-git/update-manager-image-in-git.yaml

```diff
@@ -0,0 +1,97 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: update-manager-image-in-git
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "release, tenant"
+spec:
+  description: |
+      Updates the image line in the manager yaml files in the internal-services/manager directory.
+      If mode is `pr`, a pull request is created for the update. If mode is `push`, the change is pushed
+      directly.
+  params:
+    - name: mode
+      type: string
+      description: Whether the task should create a pull request or directly push. Options are [pr, push]
+    - name: repoBranch
+      type: string
+      description: The branch in the repo to target
+      default: main
+    - name: repoUrl
+      type: string
+      description: The repo to update, starting with github.com, without https:// (e.g. github.com/org/repo).
+      default: github.com/hacbs/release/app-interface-deployments.git
+    - name: githubSecret
+      type: string
+      description: The secret containing a `token` key with value set to the GitHub access token
+    - name: image
+      type: string
+      description: The manager image to update to
+  steps:
+    - name: update-repo
+      image: quay.io/konflux-ci/release-service-utils:9089cafbf36bb889b4b73d8c2965613810f13736
+      env:
+        - name: GITHUB_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.githubSecret)
+              key: token
+        - name: EMAIL
+          valueFrom:
+            secretKeyRef:
+              name: $(params.githubSecret)
+              key: email
+        - name: NAME
+          valueFrom:
+            secretKeyRef:
+              name: $(params.githubSecret)
+              key: name
+      script: |
+          #!/usr/bin/env bash
+          set -exo pipefail
+
+          case "$(params.mode)" in
+            pr)
+              PR=true
+              ;;
+            push)
+              ;;
+            *)
+              echo "Invalid mode parameter. Only 'pr' and 'push' are allowed."
+              exit 1
+              ;;
+          esac
+
+          # Switch to /tmp to avoid filesystem permission issues
+          cd /tmp
+
+          set +x
+          git clone "https://oauth2:$GITHUB_TOKEN@$(params.repoUrl)" gitRepo
+          set -x
+          cd gitRepo
+          git config --global user.name "$NAME"
+          git config --global user.email "$EMAIL"
+          git checkout "$(params.repoBranch)"
+
+          BRANCH=$(params.repoBranch)
+          if [ -n "$PR" ] ; then
```

Comment:
I suggested to use `set -u` elsewhere. Obviously this would be an issue here. So I'd suggest to set PR=false for mode=push. Or maybe we don't even need that and can just use the param? `if [ "$(params.mode)" = "pr" ]; then`

Reply:
I added `set -u` and went with the latter (no `PR` variable)